symlink/eval_symlink.go

```diff
@@ -0,0 +1,12 @@
+package symlink
+
+// EvalSymlinks returns the path name after the evaluation of any symbolic
+// links.
+// The original implementation can be referenced to filepath.EvalSymlinks,
+// but it will return the current path other then ENOENT error while walkSymlinks.
+// If path is relative the result will be relative to the current directory,
+// unless one of the components is an absolute symbolic link.
+// EvalSymlinks calls Clean on the result.
+func EvalSymlinks(path string) (string, error) {
```

Comment:
How about keeping in `fs` package and don't export this function? We don't want outside users depends on this function.